api/opencl_runtime_layer.asciidoc

```diff
@@ -7219,6 +7219,17 @@ Otherwise, it returns one of the following errors:
     by the OpenCL implementation on the device.
   * {CL_OUT_OF_HOST_MEMORY} if there is a failure to allocate resources
     required by the OpenCL implementation on the host.
+
+When {clSetKernelArg} returns an error code different from {CL_SUCCESS}, the
+internal state of _kernel_ may only be modified when that error code is
+{CL_OUT_OF_RESOURCES} or {CL_OUT_OF_HOST_MEMORY}. When the internal state
+of _kernel_ is modified, it is implementation-defined whether:
+
+  * The argument value that was previously set is kept so that it can be used in
+    further kernel enqueues.
+  * The argument value is unset such that a subsequent kernel enqueue fails with
+    {CL_INVALID_KERNEL_ARGS}. Implementations are encouraged to favour this option
```

Comment:
```suggestion
    {CL_INVALID_KERNEL_ARGS}. Implementations are encouraged to favor this option
```

Minor: If we decide to keep this sentence then I think we should use the American English spelling.  If we decide to keep it should it be a note or a footnote instead?